### PR TITLE
Refactor duplicate CLI options

### PR DESCRIPTION
**Pull request by madrob**

### CODE REVIEW

**Review by epugh — APPROVED** [review]
Nice clean up.

**Comment by epugh** [comment]
This looks like a nice clean up, do you want me to test out the code and see if I can fix the checks?   Is there anything  you found that holds up merging this?

**Comment by epugh** [comment]
Looks like we need a JIRA ;-)

**Review by risdenk — APPROVED** [review]

**Comment by risdenk** [comment]
Merged main to this. should rerun tests. This is a good cleanup to get in.

File: solr/core/src/java/org/apache/solr/util/SolrCLI.java [hunk 1]
```diff
@@ -438,16 +459,19 @@ private static Options getToolOptions(Tool tool) {
...
 
-    if (rhs != null) {
-      for (Option opt : rhs) options.add(opt);
+    if (rhs == null) {
+      return lhs;
     }
 
-    return options.toArray(new Option[0]);
+    Option[] options = new Option[lhs.length + rhs.length];
+    System.arraycopy(lhs, 0, options, 0, lhs.length);
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/java/org/apache/solr/util/SolrCLI.java:471]: `System.arraycopy` is recommended in a few places by intellij inspections. there are some manual array copies throughout the Solr code base that could be fixed.

**Pull request merged**